x/qgb/abci_test.go

```diff
@@ -3,75 +3,137 @@ package qgb_test
+
+	// run abci methods after chain init
+	staking.EndBlocker(input.Context, input.StakingKeeper)
+	qgb.EndBlocker(ctx, *pk)
+
+	// current attestation expectedNonce should be 1 because a valset has been emitted upon chain init.
+	currentAttestationNonce := pk.GetLatestAttestationNonce(ctx)
+	require.Equal(t, uint64(1), currentAttestationNonce)
+
+	input.Context = ctx.WithBlockHeight(ctx.BlockHeight() + 1)
+	msgServer := stakingkeeper.NewMsgServerImpl(input.StakingKeeper)
 
-	// Run the staking endblocker to ensure valset is set in state
+	newOrchAddr := sdk.AccAddress(ed25519.GenPrivKey().PubKey().Address())
+	editMsg := stakingtypes.NewMsgEditValidator(
+		testutil.ValAddrs[1],
+		stakingtypes.Description{},
+		nil,
+		nil,
+		&newOrchAddr,
+		nil,
+	)
+	_, err := msgServer.EditValidator(input.Context, editMsg)
+	require.NoError(t, err)
 	staking.EndBlocker(input.Context, input.StakingKeeper)
 	qgb.EndBlocker(input.Context, *pk)
+	input.Context = ctx.WithBlockHeight(ctx.BlockHeight() + 10)
 
-	assert.NotEqual(t, currentValsetNonce, pk.GetLatestAttestationNonce(ctx))
+	// not incrementing the nonce because a change in orch address shouldn't
+	// cause a change in the validator set, from the QGB perspective.
+	assert.Equal(t, currentAttestationNonce, pk.GetLatestAttestationNonce(ctx))
 }
 
-func TestValsetEmission(t *testing.T) {
+func TestValsetCreationWhenEditingEVMAddr(t *testing.T) {
 	input, ctx := testutil.SetupFiveValChain(t)
 	pk := input.QgbKeeper
 
-	// EndBlocker should set a new validator set
+	// run abci methods after chain init
+	staking.EndBlocker(input.Context, input.StakingKeeper)
 	qgb.EndBlocker(ctx, *pk)
 
-	require.Equal(t, uint64(1), pk.GetLatestAttestationNonce(ctx))
-	attestation, found, err := pk.GetAttestationByNonce(ctx, 1)
-	require.Nil(t, err)
-	require.True(t, found)
-	require.NotNil(t, attestation)
-	require.Equal(t, uint64(1), attestation.GetNonce())
+	// current attestation expectedNonce should be 1 because a valset has been emitted upon chain init.
+	currentAttestationNonce := pk.GetLatestAttestationNonce(ctx)
+	require.Equal(t, uint64(1), currentAttestationNonce)
 
-	// get the valsets
-	require.Equal(t, types.ValsetRequestType, attestation.Type())
-	vs, ok := attestation.(*types.Valset)
-	require.True(t, ok)
-	require.NotNil(t, vs)
+	input.Context = ctx.WithBlockHeight(ctx.BlockHeight() + 1)
+	msgServer := stakingkeeper.NewMsgServerImpl(input.StakingKeeper)
+
+	newEVMAddr, err := teststaking.RandomEVMAddress()
+	require.NoError(t, err)
+	editMsg := stakingtypes.NewMsgEditValidator(
+		testutil.ValAddrs[1],
+		stakingtypes.Description{},
+		nil,
+		nil,
+		nil,
+		newEVMAddr,
+	)
+	_, err = msgServer.EditValidator(input.Context, editMsg)
+	require.NoError(t, err)
+	staking.EndBlocker(input.Context, input.StakingKeeper)
+	qgb.EndBlocker(input.Context, *pk)
+	input.Context = ctx.WithBlockHeight(ctx.BlockHeight() + 10)
+
+	assert.Equal(t, currentAttestationNonce+1, pk.GetLatestAttestationNonce(ctx))
```

Comment:
Yes, after a new validator set is created, the nonce is incremented.

Reply:
[no change needed] it seems like this test is trying verifying a side effect because it's inspecting the attestation nonce instead of the valset. In other words, what if an unrelated action incremented the attestation nonce but the valset remained the same.

Reply:
because an incrementation should only happen in two cases:

- new valset 
- new data commitment

Since we're only increasing the block height by 10: `ctx.WithBlockHeight(ctx.BlockHeight() + 10)`, then, no data  commitment should be created.

Thus,  only the a valset would increment.